File: 프로그래머스/2/42842. 카펫/카펫.py
import math

def solution(brown, yellow):
    mult = brown + yellow
    add = (brown/2+2)
    # a*b, a+b x^2 + (a+b) x + ab = x^2 + add*x + mult. (-add+- sqrt(add^2-4*mult))//2
    
    a = -add/2
    b = math.sqrt(add**2 - 4*mult)/2
    return [-a+b,-a-b]

File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const wPlusH = brown/2+2;
    const wMultH = yellow + brown;

    const w = wPlusH/2 + ((wPlusH/2)**2-wMultH)**0.5
    const h = wPlusH/2 - ((wPlusH/2)**2-wMultH)**0.5;
    return [w,h]
}